### PR TITLE
Feature/test attributes

### PR DESCRIPTION
**Pull request by NikoBergemann**
### Reason for this PR
- groups should not be able to be manipulated by everyone

### Changes in this PR
- groups can only be manipulated by users that have hf.Attribute "sysAdmin=true"

should not be released until https://github.com/upb-uc4/hlf-network/pull/118 is merged

### CODE REVIEW

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 1]
```diff
@@ -56,6 +56,7 @@ public String addUserToGroup(final Context ctx, String enrollmentId, String grou
 
         try {
             cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId, groupId});
+            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add("sysAdmin");}});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:59]: Maybe change the order, first validate attributes and then validate approvals?

**Review by anemonekk — APPROVED** [review]

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java (hunk 1 above)
Comment (matthias-geuchen) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:59]: "sysAdmin" should be stored in some static final variable, like groups (like group names in AccessManager)

**Review by matthias-geuchen — CHANGES_REQUESTED** [review]

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java (hunk 1 above)
Comment (NikoBergemann) [reply UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java]: resolved in new commits

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 2]
```diff
@@ -56,6 +57,7 @@ public String addUserToGroup(final Context ctx, String enrollmentId, String grou
 
         try {
             cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId, groupId});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:59]:
```suggestion
            
```

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 3]
```diff
@@ -56,6 +57,7 @@ public String addUserToGroup(final Context ctx, String enrollmentId, String grou
...
+            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add(AccessManager.HLF_ATTRIBUTE_SYSADMIN);}});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:60]:
```suggestion
            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add(AccessManager.HLF_ATTRIBUTE_SYSADMIN);}});
            cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId, groupId});
```

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 4]
```diff
@@ -97,6 +99,7 @@ public String removeUserFromGroup(final Context ctx, String enrollmentId, String
 
         try {
             cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId, groupId});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:101]:
```suggestion
            
```

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 5]
```diff
@@ -97,6 +99,7 @@ public String removeUserFromGroup(final Context ctx, String enrollmentId, String
...
+            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add(AccessManager.HLF_ATTRIBUTE_SYSADMIN);}});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:102]:
```suggestion
            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add(AccessManager.HLF_ATTRIBUTE_SYSADMIN);}});
            cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId, groupId});
```

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 6]
```diff
@@ -129,6 +132,7 @@ public String removeUserFromAllGroups(final Context ctx, String enrollmentId) {
         ChaincodeStub stub = ctx.getStub();
         try {
             cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:134]:
```suggestion
            
```

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java [hunk 7]
```diff
@@ -129,6 +132,7 @@ public String removeUserFromAllGroups(final Context ctx, String enrollmentId) {
...
+            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add(AccessManager.HLF_ATTRIBUTE_SYSADMIN);}});
```
Comment (anemonekk) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java:135]:
```suggestion
            cUtil.validateAttributes(ctx, new ArrayList<String>() {{add(AccessManager.HLF_ATTRIBUTE_SYSADMIN);}});
            cUtil.validateApprovals(ctx, contractName, transactionName, new String[]{enrollmentId});
```

**Review by anemonekk — APPROVED** [review]

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java (hunk 1 above)
Comment (NikoBergemann) [reply UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/group/GroupContract.java]: resolved in new updates

File: UC4-chaincode/src/test/java/de/upb/cs/uc4/chaincode/util/TestUtil.java [hunk 8]
```diff
@@ -35,6 +35,8 @@ public static Context mockContext(MockChaincodeStub stub, String clientId) {
         stub.setCurrentId(clientId);
         ClientIdentity testId = mock(ClientIdentity.class);
         when(testId.getId()).thenReturn(clientId);
+        when(testId.getAttributeValue("sysAdmin")).thenReturn("true");
+        when(testId.assertAttributeValue("sysAdmin", "true")).thenReturn(true);
```
Comment (matthias-geuchen) [inline UC4-chaincode/src/test/java/de/upb/cs/uc4/chaincode/util/TestUtil.java:39]:
```suggestion
        when(testId.getAttributeValue(AccessManager.HLF_ATTRIBUTE_SYSADMIN)).thenReturn("true");
        when(testId.assertAttributeValue(AccessManager.HLF_ATTRIBUTE_SYSADMIN "true")).thenReturn(true);
```

**Review by matthias-geuchen — CHANGES_REQUESTED** [review]

File: UC4-chaincode/src/test/java/de/upb/cs/uc4/chaincode/util/TestUtil.java [hunk 9]
```diff
@@ -35,6 +35,8 @@ public static Context mockContext(MockChaincodeStub stub, String clientId) {
         stub.setCurrentId(clientId);
         ClientIdentity testId = mock(ClientIdentity.class);
         when(testId.getId()).thenReturn(clientId);
+        when(testId.getAttributeValue(AccessManager.HLF_ATTRIBUTE_SYSADMIN)).thenReturn("true");
+        when(testId.assertAttributeValue(AccessManager.HLF_ATTRIBUTE_SYSADMIN "true")).thenReturn(true);
```
Comment (NikoBergemann) [inline UC4-chaincode/src/test/java/de/upb/cs/uc4/chaincode/util/TestUtil.java:39]:
```suggestion
        when(testId.assertAttributeValue(AccessManager.HLF_ATTRIBUTE_SYSADMIN, "true")).thenReturn(true);
```

**Review by matthias-geuchen — APPROVED** [review]

**Pull request merged**